### ISSUE TITLE
Advanced Filter GUI displaying incorrectly

### ISSUE THREAD

**Issue by dirtyfargo** [gr8pefish/IronBackpacks#42]
Description: GUI button options for the advanced filter are appearing over the basic filter slots as shown here: https://gyazo.com/016c256b588bcf5847b3716f85e9ffa2
However, it seems that both the basic and advanced slots are able to acquire the 'advanced filtering' properties.
Version: 1.2.13 (But I've had this bug for a good few versions now)
Forge: 10.13.4.1614

Side note: Has a 'void filter' or 'trash filter' been considered for a possible backpack upgrade in the future? I would definitely use that one =)

**Comment by gr8pefish** [comment]
Haha nice backpack name ;) I'll look into the bug and fix it. Curious as to why you didn't report it earlier, (if you had it for a few versions).

And yes, that is a good idea, a void filter. I will add that in.

**Comment by dirtyfargo** [comment]
I honestly just completely forgot about it if I'm honest lol, I went to re-do/improve my backpack filters for mining today and then saw it and remembered. Thanks haha, it took every ounce of my creativity ;P
And yay I shall be able to void all the things into nothingness!
Thanks a bunch :)

**Comment by gr8pefish** [comment]
Done, should be on 1.2.14 on curse momentarily. I will make that void upgrade for next release, but I wanted to get the fix out to you asap! 

Any other suggestions/comments/concerns by the way? You know how desperate I am for input. xD

**Comment by dirtyfargo** [comment]
I feel like I always have decent ideas while playing then forget them afterwards when it matters :P I need to write stuff down! Off the top of my head though, maybe giving the backpacks the ability to be placed in a baubles slot and still keep its functionality, could even be a 'Bauble' upgrade that can be disabled via cfg if you don't play with Thaumcraft. 
Another idea, what about an upgrade that would allow you to link the backpack to 1 specific inventory ( a chest at your base for example) and have a filter gui that would automatically send the items you choose to it.  This way you could send ores or whatever to your processing machines and organized storage system auto-magically. Not sure if this is too overpowered or not haha, but I would love using that.
Thanks for the fast fix by the way I will be testing it out asap! =)

**Comment by gr8pefish** [comment]
You can use Bibliocraft's clipboard to keep notes in-game. Or just something like notepad to keep notes on your computer. ;)
- Baubles/Traveler's Gear stuff is being considered. I'm not sure how feasible it is from a coding standpoint yet. Plus I don't know if it would apply well. Does it go in the belt slot? That just seems odd. And if I did implement it, would it be mandatory and I would disable my own equipping? Or is my way with keybindings better? Because I can't have both simultaneously, or you would be able to "equip" two backpacks at once. I agree it could be cool, it is just a bit of a thorny issue I have to sort through first.
- Yeah I have also been thinking about some remote upgrade. My current idea is to simply have another backpack that is an ender one (that links to your ender chest) that can accept the filters, and just leave it at that. To do that I want to make my API first though, as it will make it much faster, so I have some work to do before that. However, your idea of a specified filter linking to a specific chest is definitely interesting, I will have to think about that one...

Thanks for the ideas!

**Comment by dirtyfargo** [comment]
- Thanks for the tip! I have actually started to take notes on a ton of things just by hand actually, although I should try out the clipboard :P
- Yeah that makes sense about baubles, and ultimately it's not necessary anyways.
- That's a neat idea about the ender chest back pack :) I guess the nice quality of being able to link it to any specific chest is that you could have multiple different item extraction setups instead of being limited to just the single ender chest. I think both could be helpful additions for sure.

No problem, it's nice to feel like I'm helping honestly!

**Comment by gr8pefish** [comment]
Cool, I took notes of your comments. A new version is up on curse that adds the void filter!

And you are helpful! :)

**Comment by dirtyfargo** [comment]
Ahhh perfect I was literally just updating my modpack haha yay :)
Thank you kindly!

**Comment by dirtyfargo** [comment]
After playing with the void filter I'm realizing certain things that I would want to use it for (such as swords and bows from mob drops) can have a great range of different damage values, and would require the fuzzy matching selection from the advanced filter upgrade gui to dispose of them all. Would it be possible to perhaps change the recipe to require an advanced filter instead of a basic one, and provide those options? Or perhaps a separate new upgrade providing a more advanced void filter?
Just thoughts while playing :)

**Comment by gr8pefish** [comment]
Hmm I don't want to implement another advanced filter-type gui upgrade right now, that is a lot of work. Eventually maybe, but not now. However, for now, would you recommend I just change the void filter to detect the items in it's filter in a fuzzy manner? That is a super easy change.

**Comment by dirtyfargo** [comment]
That would work, I can't think of any foreseeable issues with that method of doing it anyways! I just get so many bows from skeletons, and swords from pigmen and wither skeletons that I'm looking for a way to delete them all automatically so as not to clog up my inventory :)

**Comment by gr8pefish** [comment]
Okay will do! I'm busy on 1.8 stuff, so I think I'm going to wait on pushing the next 1.7.10 version of IB until I have more fixes in it, so don't expect it immediately ;)

Edit: I also acquired someone to upgrade my textures, so I'll probably push the update out when he gets finished with that (unless it takes forever).

**Comment by dirtyfargo** [comment]
Ooo new textures that is exciting =)
Sounds good and thank you very much once again!
